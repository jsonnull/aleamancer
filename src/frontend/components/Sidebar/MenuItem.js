// @flow
import React from 'react'
import styled from 'styled-components'
import { colors } from 'frontend/styles/common'
import Tooltip from '../Tooltip'

const Item = styled.div`
  text-align: center;
  height: 4.8rem;
  width: 4.8rem;
  line-height: 4.8rem;
  background-repeat: no-repeat;
  background-size: ${props => (props.selected ? '100% 4px' : '0 4px')};
  background-position: ${props => (props.selected ? '0 4.4rem' : '50% 4.4rem')};
  background-image: linear-gradient(black 0%, black 100%);
  transition: background-size 100ms, background-position 100ms;

  &:hover {
    background-color: ${props =>
      props.theme.name === 'dark'
        ? props.theme.backgroundSecondary
        : colors.lightBlue};
    color: white;
    cursor: pointer;
  }
`

type Props = {
  name: string,
  icon: string,
  selected: boolean,
  action: Function
}

const MenuItem = (props: Props) => {
  const { name, icon, selected, action } = props

  return (
    <Tooltip content={name} placement="bottom">
      <Item selected={selected} onClick={() => action(name)}>
        <i className={`fa ${icon}`} />
      </Item>
    </Tooltip>
  )
}

export default MenuItem
