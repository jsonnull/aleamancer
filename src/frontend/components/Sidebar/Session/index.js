// @flow
import React from 'react'

type Props = {}

export default class Session extends React.Component<Props> {
  render() {
    return <div />
  }
}
