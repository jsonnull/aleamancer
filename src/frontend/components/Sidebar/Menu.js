// @flow
import React from 'react'
import styled from 'styled-components'
import type { Tab } from 'common/types'
import MenuItem from './MenuItem'

const MenuContainer = styled.div`
  display: flex;
  flex-direction: row;
  height: 4.8rem;
  color: black;
  background-color: ${props =>
    props.theme.name === 'dark' ? props.theme.backgroundSecondary : ''};
`

type MenuButton = [Tab, string] | 'separator'

const buttons: Array<MenuButton> = [
  ['Session', 'fa-globe'],
  ['Character', 'fa-id-card-o']
]

type Props = {
  tab: Tab,
  changeTab: Function
}
class Menu extends React.Component<Props> {
  render() {
    return (
      <MenuContainer>
        {buttons.map(button => {
          const [name, icon] = button
          const selected = name == this.props.tab

          return (
            <MenuItem
              key={name}
              name={name}
              icon={icon}
              selected={selected}
              action={this.props.changeTab}
            />
          )
        })}
      </MenuContainer>
    )
  }
}

export default Menu
