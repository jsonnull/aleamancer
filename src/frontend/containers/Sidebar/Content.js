// @flow
import React from 'react'
import type { Tab } from 'common/types'
import Session from './Session'

type Props = {
  tab: Tab
}
const Content = (props: Props) => {
  const { tab } = props

  if (tab === 'Session') {
    return <Session />
  }

  return null
}

export default Content
