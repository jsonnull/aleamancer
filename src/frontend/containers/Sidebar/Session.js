// @flow
import { connect } from 'react-redux'
import Session from 'frontend/components/Sidebar/Session'
import sessionNameSelector from 'frontend/selectors/sessionName'
import type { State } from 'frontend/store'

type StateProps = { name: string }
const mapStateToProps = (state: State): StateProps => ({
  name: sessionNameSelector(state)
})

export default connect(mapStateToProps)(Session)
