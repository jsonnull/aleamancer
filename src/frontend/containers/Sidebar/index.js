// @flow
import { compose } from 'recompose'
import { connect } from 'react-redux'
import { changeSidebarTab } from 'frontend/actions'
import Sidebar from 'frontend/components/Sidebar'
import sessionIdSelector from 'frontend/selectors/sessionId'
import { graphql } from 'react-apollo'
import gql from 'graphql-tag'
import type { State } from 'frontend/store'

const mapStateToProps = (state: State) => ({
  sessionId: sessionIdSelector(state),
  open: state.sidebar.open,
  tab: state.sidebar.tab
})

const mapDispatchToProps = {
  changeTab: changeSidebarTab
}

const currentSession = gql`
  query sessionName($id: ID!) {
    game(id: $id) {
      name
    }
  }
`

export default compose(
  connect(mapStateToProps, mapDispatchToProps),
  graphql(currentSession, {
    name: 'currentSession',
    options: ownProps => ({
      variables: {
        id: ownProps.sessionId
      }
    })
  })
)(Sidebar)
