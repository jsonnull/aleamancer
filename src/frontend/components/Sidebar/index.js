// @flow
import React from 'react'
import styled from 'styled-components'
import Content from 'frontend/containers/Sidebar/Content'
import type { Tab } from 'common/types'
import { colors, fontSize, fonts } from 'frontend/styles/common'
import Menu from './Menu'

const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 300px;
  display: flex;
  flex-direction: column;
`

const Top = styled.div`
  background: ${colors.lightGray};
`

const Header = styled.h1`
  font-family: ${fonts.heading};
  font-size: ${fontSize.large};
  line-height: 1;
  padding: 1rem 1rem 0;
  color: ${props => props.theme.color};
  margin: 0;
`

const ContentContainer = styled.div`
  display: flex;
  flex-direction: column;
  flex: 1;
`

type Props = {
  currentSession: {
    game: {
      name: string
    }
  },
  open: boolean,
  tab: Tab,
  changeTab: Function
}

const Sidebar = (props: Props) => {
  const { tab, changeTab, currentSession } = props

  return (
    <Container>
      <Top>
        <Header>{currentSession.game && currentSession.game.name}</Header>
        <Menu tab={tab} changeTab={changeTab} />
      </Top>
      <ContentContainer>
        <Content tab={tab} />
      </ContentContainer>
    </Container>
  )
}

export default Sidebar
