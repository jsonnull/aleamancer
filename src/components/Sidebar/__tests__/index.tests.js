// @flow
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Sidebar from '../index.js'

describe('Sidebar component', () => {
  const renderer = new ShallowRenderer()
  it('renders correctly', () => {
    renderer.render(
      <Sidebar name="name" open={true} tab="Session" changeTab={() => {}} />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})
