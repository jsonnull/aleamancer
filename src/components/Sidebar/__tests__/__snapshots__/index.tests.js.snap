// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component renders correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.h1>
      name
    </styled.h1>
    <Menu
      changeTab={[Function]}
      tab="Session"
    />
  </styled.div>
  <styled.div>
    <Content
      tab="Session"
    />
  </styled.div>
</styled.div>
`;
